esp/bindings/SOAP/Platform/soapparam.hpp

```diff
@@ -1022,7 +1022,8 @@ class SoapEnumParamNew
 
     void marshall(IRpcMessage &rpc_call, const char *tagname, const char* itemname, const char* elementtype="",const char *basepath="", const char *prefix="")
     {
-        rpc_call.add_value(basepath, prefix, tagname, "string", enumstrings.item(value));
+        if ((value >= 0) && (value < enumstrings.length()))
```

Comment:
Not sure what type is, but testing for >= 0 may generate a "redundant comparison" warning if unsigned

Reply:
Should an enum always be unsigned? The existing code uses '-1' as a flag for the value uninitialized. We may add a special flag for the value uninitialized. But, what if an int type is passed in? @afishbeck any thought?

Reply:
A workaround ?
if (enumstrings.isItem(value))    rpc_call.add_value(basepath, prefix, tagname, "string", enumstrings.item(value));

Reply:
These values are basically int.. which represents the internal index of the allowed value.

These ESDL enums are similar enums in XML schemas.. that is they are more like a list of allowed values for the markup (xml, json, etc) tag.  Each allowed value is given an index which becomes the c++ enum value.  This is basically an int that maps to a string or another (external facing) xsd type.

Although the implementation could be cleaned up because we rely on the default type for a simple c++ enum declaration.